lib/machineparams.py

```diff
@@ -1,66 +1,51 @@
-import os
-import sys
-from abc import ABC, abstractmethod
-
-class ParentMachine(ABC):
-
+class ParentMachine(object):
+    '''Parent class representing an empty machine'''
     def __init__(self):
-        pass
-
-    @abstractmethod
-    def parameters(params):
-        pass
+        self.params = {
+            'inj':{
+                'rfvol': None,
+                'sigz' : None,
+                'sige' : None
+            },
+            'col':{
+                'rfvol': None,
+                'sigz' : None,
+                'sige' : None
+            }
+        }
+        return
+    def parameters(self, runtype):
+        return self.params[runtype]
 
 class LHC(ParentMachine):
-
     def __init__(self):
         ParentMachine.__init__(self)
-
-    def parameters(self, params):
-        inj = {}
-        inj['RF_voltage'] = 8.0 #[MV]
-        inj['sigz'] = 0.11 #[mm]
-        inj['sige'] = 4.5e-04 #[]
-
-        col = {}
-        col['RF_voltage'] = 16.0 #[MV]
-        col['sigz'] = 0.77e-1 #[mm]
-        col['sige'] = 1.1e-04 #[]
-
-        params['inj'] = inj
-        params['col'] = col
+        
+        self.params['inj']['rfvol'] = 8.0 #[MV]
+        self.params['inj']['sigz'] = 0.11 #[mm]
+        self.params['inj']['sige'] = 4.5e-04 #[]
+        
+        self.params['col']['rfvol'] = 16.0 #[MV]
+        self.params['col']['sigz'] = 0.77e-1 #[mm]
+        self.params['col']['sige'] = 1.1e-04 #[]
 
 class HLLHC(ParentMachine):
-
     def __init__(self):
         ParentMachine.__init__(self)
-
-    def parameters(self, params):
-        inj = {}
-        inj['RF_voltage'] = 8.0 #[MV]
-        inj['sigz'] = 0.11 #[mm]
-        inj['sige'] = 4.5e-04 #[]
-
-        col = {}
-        col['RF_voltage'] = 16.0 #[MV]
-        col['sigz'] = 0.77e-1 #[mm]
-        col['sige'] = 1.1e-04 #[]
-
-        params['inj'] = inj
-        params['col'] = col
-
-class Machine(object):
-
+        
+        self.params['inj']['rfvol'] = 8.0 #[MV]
+        self.params['inj']['sigz'] = 0.11 #[mm]
+        self.params['inj']['sige'] = 4.5e-04 #[]
+        
+        self.params['col']['rfvol'] = 16.0 #[MV]
+        self.params['col']['sigz'] = 0.77e-1 #[mm]
+        self.params['col']['sige'] = 1.1e-04 #[]
+        
+class MachineConfig(object):
     def __init__(self, machine, runtype):
-        self.runtype = runtype
-        self.params = {}
         if machine.lower() == 'lhc':
-            LHC().parameters(self.params)
+            return LHC().parameters(runtype)
         elif machine.lower() == 'hllhc':
-            HLLHC().parameters(self.params)
+            return HLLHC().parameters(runtype)
```

Comment:
This is a mistake, the constructor can not return dict (actually anything else except None).

Reply:
wait a second: here python will first call the constructor, then the `.parameters()` method.
If you want, the mistake is when using the `MachineConfig` class, do you refer to this?

Reply:
That's not the problem. 
The problem is that __init__() can not return "things" except None. 
So we should not put a return statement in __init__() method.

Reply:
ok - what about something like (indentation may not be perfect):
```
class ParentMachine(object):
    [...]
    def parameters(self, runtype=None):
        if ( runtype is None ):
            return self.params
        else:
￼            return self.params[runtype]
class MachineConfig(object):
￼     def __init__(self, machine):
         self.parameters={}
￼         if machine.lower() == 'lhc':
￼             self.parameters=LHC().parameters()
￼         elif machine.lower() == 'hllhc':
￼             self.parameters=HLLHC().parameters()
    def parameters(self, runtype=None):
        if ( runtype is None ):
         return self.params
        else:
￼         return self.params[runtype]
```

Reply:
Ok, it works! 
But  base on the usage in class `MacineConfig`  we get the parameters dict through `LHC().parameters()` , the statements in `parameters` method of `ParentMachine` class don't make sense. maybe just return the params dict directly. 
Anyway, it works, I will create a new pull request to update the code.

Reply:
well, yes, `ParentMachine.parameters()` and `MachineConfig.parameters()` are basically identical, and the code is not super elegant because of this.
The first method is used by both the `LHC` and the `HLLHC` classes because of inheritance. Maybe, the `MachineConfig` class should inherit from `ParentMachine` as well, what do you think?

Reply:
Yes, it's an option, but it will mess up the inheritance. The `ParentMachine` class stand for a 'machine', but `MachineConfig` is not a machine. 
It's not a complicated module, maybe we just modify the `parameters` method in `ParentMachine` or `MachineConfig` slightly, or just keep it as it is.

Reply:
ok let's keep it like this